Task 1: Improved UNET on ISICs - 45820188

AndrewLuong6 opened the pull request:
Hi Shakes & Teaching Team,

I've implemented the Improved UNET on the ISICs 2018 Dataset. ~~There were some implementation issues, and as such, the accuracy is around 60%.~~ Accuracy reached around 86% average.

The files included in this are **driver.py** and **model.py** which include the code for the Improved UNET model, as well as the driver code to run the images through this model. There is also a README that describes the problem, explains the model and shows some sample output. The images used for the README are stored in the Images directory.

If there are any issues or changes that you require, feel free to let me know. 

Thank you!
Andrew Luong (45820188)

AndrewLuong6 committed:
First commit. Begun image processing on a subset of images. Incomplete model included

AndrewLuong6 committed:
Started to build draft of Improved UNET Model.

AndrewLuong6 committed:
Modified Data Processing Method. Started training model with data - however accuracy is still wrong. Also output a predictions vs expected plot

AndrewLuong6 committed:
Extracted UNET Model from Jupyter Notebook into Model.py File

AndrewLuong6 committed:
Added driver.py file from original notebook. Removed notebook from Git
repo

AndrewLuong6 committed:
Added comments to model.py and fixed stride bug

Added docstring comments and general comments to model.py file.  Also fixed a possible bug with the segmentation layers having incorrect kernel sizes.

AndrewLuong6 committed:
Added comments to driver.py

Added docstring comments to driver as well as moving general function calls into a main function.

AndrewLuong6 committed:
Created README

Created a README file without images currently. Also made some minor changes inside the driver file regarding predictions output.

AndrewLuong6 committed:
Fixed bugs in UNET Model

Fixed some bugs within the last few convolution layers of the model.
Added an output heading within the README.
Added the output of the model's loss, accuracy and dice coefficient.

AndrewLuong6 committed:
Added Images to README

Added some images to the README file, with some extra discussion on the output accuracy.

AndrewLuong6 committed:
Fixed README Image Syntax

Wrong paths to images was used. Fixed to be absolute paths within the repo.

AndrewLuong6 committed:
Fixed last README Image Issue

(Hopefully) Fixed the last of README image formatting.

wooboyeong commented:
Code
* Dice implemented but not working as expected
* Could be an issue with image preprocessing e.g. Did you normalise the images?

README
* Please report average DSC for test set in README

General Comments
* Commit messages OK
* Pull request OK

AndrewLuong6 committed:
Added Accuracy Plotting and Adjusted Hyper Params

Added functions to plot the dice coefficient and loss values.

Adjusted hyper parameters to be smaller image size, allowing for more epochs.

Normalised the images by diving by 255.

Also added the output log of 1000 epochs for proof.

AndrewLuong6 committed:
Update README.md

Fixed typos in README images and formatting.

AndrewLuong6 committed:
Updated Images to Include Background

Previous images had clear background. Added white background so text could be read.

AndrewLuong6 committed:
Update README.md

Clarified where the dice coefficient value was computed from.

shakes76 commented:
## TF/Torch Usage
Adequate use and implementation
## Good Practice (Design/Commenting)
Good spacing and comments
Header blocks
## Algorithm 
Driver Script present
Shows Usage & Demo & Visualisation & Data usage 
Module present 
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
Good Description and Comments 
Markdown used PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No feedback required
Request Description OK
